### PR TITLE
Bump StoryBook to ^7.0.26

### LINKED ISSUE #1947: Bump Storybook to v7
This issue is all about bumping Storybook to the new version 7.

The basic approach would be to only run `npx storybook upgrade` but it won't work for this major release.

Please use [their migration guide](https://storybook.js.org/docs/react/migration-guide) to do it (see also https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#from-version-65x-to-700 that might be useful).

Maybe the easiest and more precise way would be to drop everything regarding Storybook configuration and then reinstall everything from scratch. I'm saying that because there are a lot of breaking changes, and the one regarding Webpack 4 could be already impactful for us.

On top of these tasks:
- Check if some new features could be useful for us
- Double-check that https://github.com/Orange-OpenSource/ods-storybook-theme is still working without any modifications
- Try to bump `sass-loader` as well (or remove it if no more useful) which was stuck in this version so far

### PR DESCRIPTION
**Pull request by louismaximepiton**
_Note: Please transform `- [ ]` into `- (NA)` in the description when things are not applicable_
 
 ### Related issues
 
 Closes #1947.
Fixes some issues in #1626.
 
 ### Description
 
 Bump Storybook dependency to 7.0.26.
⚠️ Updated the node version of Netlify from 14 to 16 since it requires node 16 at least to run
 
 ### Motivation & Context
 
 Be up to date and not left behind.
 
 ### Types of change
 
 
 
 
 - New feature (non-breaking change which adds functionality)
 
 ### Live previews
 
 
 
 - <https://deploy-preview-2103--boosted.netlify.app/storybook>
 
 ### Checklist
 
 
 
 
 #### Contribution
 
 - [x] I have read the [contributing guidelines](https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/blob/main/.github/CONTRIBUTING.md)
 
 #### Accessibility
 
 - (NA) My change follows accessibility good practices; I have at least run axe
 
 #### Design
 
 - (NA) My change respects the design guidelines defined in [Orange Design System](https://system.design.orange.com/0c1af118d/p/8118d1-web)
 - (NA) My change is compatible with responsive display
 
 #### Development
 
 - [x] My change follows the [developer guide](https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/wiki/Developer-guide)
 - (NA) I have added JavaScript unit tests to cover my changes
 - (NA) I have added SCSS unit tests to cover my changes
 
 #### Documentation
 
 - (NA) My change introduces changes to the documentation and/or I have updated the documentation accordingly
 
 ### Checklist (for Core Team only)
 
 - (NA) My change introduces changes to the migration guide
 - (NA) My new component is added in Storybook
 - (NA) My new component is compatible with RTL
 - (NA) Manually run [BrowserStack tests](https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/actions/workflows/browserstack.yml)
 - (NA) Manually test browser compatibility with BrowserStack (Chrome >= 60, Firefox >= 60 (+ ESR), Edge, Safari >= 12, iOS Safari, Chrome & Firefox on Android)
 - [ ] Re-rerun `npx @hipster/mdx2-issue-checker`.
 - [ ] Check for the Boosted theme (Lead dev review)
 - [ ] Maybe regenerate the package-lock from scratch
 
 #### After the merge
 
 - (NA) Manually launch [Percy tests](https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/actions/workflows/percy.yml)

### CODE REVIEW

File: package.json
Comment (louismaximepiton) [inline package.json:1]: Check for storybook theme

File: package.json [hunk 1]
```diff
@@ -161,6 +160,8 @@
     "pa11y-ci-reporter-html": "^5.1.1",
     "postcss": "^8.4.24",
     "postcss-cli": "^10.1.0",
+    "react": "^18.2.0",
+    "react-dom": "^18.2.0",
```
[the remark this answers is not in the document]
Comment (louismaximepiton) [reply package.json:164]: This seems to be mandatory (https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#react-peer-dependencies-required)

File: package.json [hunk 2]
```diff
@@ -95,10 +95,10 @@
     "watch-css-test": "nodemon --watch scss/ --ext scss,js --exec \"npm run css-test\"",
     "watch-js-main": "nodemon --watch js/src/ --ext js --exec \"npm-run-all js-lint js-compile\"",
     "watch-js-docs": "nodemon --watch site/assets/js/ --ext js --exec \"npm run js-lint\"",
-    "storybook": "npm run storybook-generate && start-storybook -p 6006 --no-manager-cache",
```
[the remark this answers is not in the document]
Comment (louismaximepiton) [reply package.json:98]: Removed from Storybook, there seems to be no replacement so let's keep it like this for the moment https://github.com/storybookjs/storybook/pull/20895/files

**Review by julien-deramond — COMMENTED** [review]
## Review (In Progress)

- [x] Run Storybook locally
  - [ ] Why isn't there a "Docs" tab anymore? 
- [x] Build Storybook locally and run from a local server
- [ ] Do the same but after having cloned the repository
- [ ] Create a `gh-pages` branch to check if everything's working the same way
- [ ] Check all the addons
- [ ] Check the documentation for the configuratons
- [ ] Verify `options.measure: false`
- [ ] Ask someone to run it on Windows
- [ ] Check Orange theme
- [ ] Check rendering of code blocks
- [ ] Simulate a Zeroheight integration
- [ ] Check if `--no-manager-cache` isn't an issue or could be replaced by something else
- [x] Run `npx storybook@latest upgrade`

**Comment by julien-deramond** [comment]
Haven't checked the new features/deprecations but do you know if "Docs" tab has now disappeared (with the "Show code" view)

Now:
Screenshot 2023-07-07 at 08 15 00

Before:
Screenshot 2023-07-07 at 08 15 09

**Comment by louismaximepiton** [comment]
FYI, the layout changed, it might have impacts on the DSM that should be studied. 

Source: https://github.com/storybookjs/storybook/issues/21720#issuecomment-1479083653

**Review by julien-deramond — APPROVED** [review]
Good job! LGTM! Let's deploy in `gh-pages` and double-check that there's no regression in the DSM.

**Pull request merged**